src/viewer.js

```diff
@@ -2541,11 +2542,31 @@ function onCanvasDrag( event ) {
         if( !this.panVertical ){
             event.delta.y = 0;
         }
-        this.viewport.panBy( this.viewport.deltaPointsFromPixels( event.delta.negate() ), gestureSettings.flickEnabled );
+
         if( this.constrainDuringPan ){
-            this.viewport.applyConstraints();
+            var delta = this.viewport.deltaPointsFromPixels( event.delta.negate() );
+
+            this["viewport"].centerSpringX.target.value += delta.x;
+            this["viewport"].centerSpringY.target.value += delta.y;
+
+            var bounds = this.viewport.getBounds();
+            var constrainedBounds = this.viewport.getConstrainedBounds();
+
+            this["viewport"].centerSpringX.target.value -= delta.x;
+            this["viewport"].centerSpringY.target.value -= delta.y;
+
+            if (bounds.x != constrainedBounds.x) {
+                event.delta.x = 0;
```

Comment:
I think instead of just zeroing the delta, the delta should be adjusted so the pan will exactly touch the constraint edge. Otherwise if you're dragging quickly, it might catch before hitting the edge.

Btw, it seems it might be cleaner to just do `var delta = this.viewport.deltaPointsFromPixels( event.delta.negate() );` at the top and just use delta instead of event.delta after that.